tcms/rpc/tests/test_testexecution.py

```diff
@@ -334,6 +334,34 @@ def test_update_with_non_existing_assignee_username(self):
                 "assignee": 'nonExistentUsername'
             })
 
+    def test_update_with_numeric_case_text_version(self):
+        self._update_case_text_version()
+        latest_case_text_version = self.case_run_1.case.history.latest().history_id
+
+        execution = self.rpc_client.TestExecution.update(self.case_run_1.pk, {
+            "case_text_version": str(latest_case_text_version)
+        })
+        self.assertEqual(execution["case_text_version"], latest_case_text_version)
+
+    def test_update_with_latest_case_text_version(self):
+        self._update_case_text_version()
+        latest_case_text_version = self.case_run_1.case.history.latest().history_id
+
+        execution = self.rpc_client.TestExecution.update(self.case_run_1.pk, {
+            "case_text_version": 'latest'
+        })
+        self.assertEqual(execution["case_text_version"], latest_case_text_version)
```

Comment:
now that I look at the tests, they are almost identical (except for the parameter passed to the API call), so we can do something like:
```python
    def test_update_with_numeric_case_text_version(self):
        latest_case_text_version = self.case_run_1.case.history.latest().history_id
        _test_update_case_text_version(str(latest_case_text_version))

    def test_update_with_latest_case_text_version(self):
        _test_update_case_text_version('latest')

    def _test_update_case_text_version(self, version):
        initial_case_text_version = self.case_run_1.case_text_version
        self.rpc_client.TestCase.update(self.case_run_1.case.pk, {
            'summary': 'This was updated',
            'text': 'new TC text',
        })
        self.case_run_1.case.refresh_from_db()
        latest_case_text_version = self.case_run_1.case.history.latest().history_id
        self.assertNotEqual(initial_case_text_version, latest_case_text_version)

        execution = self.rpc_client.TestExecution.update(self.case_run_1.pk, {
            "case_text_version": version
        })
        self.assertEqual(execution["case_text_version"], latest_case_text_version)

```

Reply:
For test_update_with_numeric_case_text_version I think it will differ logically because we need to pass the latest version after updating the test case and not the latest before the update @asankov

Reply:
I got it. I guess little duplication in the tests does no hurt

Reply:
FYI Copy & paste is usually preferred in tests b/c e don't want scenarios to have shared state. Ideally each one would be totally independent.